File: app/src/main/java/com/disruption/cookcentral/ui/search/SearchedRecipesAdapter.java
package com.disruption.cookcentral.ui.search;

import android.annotation.SuppressLint;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;
import androidx.recyclerview.widget.ListAdapter;
import androidx.recyclerview.widget.RecyclerView;

import com.disruption.cookcentral.databinding.SearchedRecipeItemBinding;
import com.disruption.cookcentral.models.search.SearchedRecipe;

public class SearchedRecipesAdapter extends ListAdapter<SearchedRecipe, SearchedRecipesAdapter.RecipeViewHolder> {

    private static final DiffUtil.ItemCallback<SearchedRecipe> DIFF_CALLBACK =
            new DiffUtil.ItemCallback<SearchedRecipe>() {
                @Override
                public boolean areItemsTheSame(@NonNull SearchedRecipe oldItem, @NonNull SearchedRecipe newItem) {
                    return oldItem.getId() == newItem.getId();
                }

                @SuppressLint("DiffUtilEquals")
                @Override
                public boolean areContentsTheSame(@NonNull SearchedRecipe oldItem, @NonNull SearchedRecipe newItem) {
                    return oldItem == newItem;
                }
            };

    public SearchedRecipesAdapter() {
        super(DIFF_CALLBACK);
    }

    @NonNull
    @Override
    public RecipeViewHolder onCreateViewHolder(@NonNull ViewGroup parent,
                                               int viewType) {
        SearchedRecipeItemBinding binding =
                SearchedRecipeItemBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false);
        return new RecipeViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull RecipeViewHolder holder, final int position) {
        SearchedRecipe recipe = getItem(position);
        holder.bind(recipe);
    }

    class RecipeViewHolder extends RecyclerView.ViewHolder {

        private SearchedRecipeItemBinding mRecipeBinding;

        RecipeViewHolder(SearchedRecipeItemBinding binding) {
            super(binding.getRoot());
            mRecipeBinding = binding;
        }

        private void bind(SearchedRecipe recipe) {
            mRecipeBinding.setRecipe(recipe);
        }
    }
}

File: app/src/main/java/com/disruption/cookcentral/ui/search/SearchedRecipesAdapter.kt
package com.disruption.cookcentral.ui.search

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import com.disruption.cookcentral.databinding.SearchedRecipeItemBinding
import com.disruption.cookcentral.models.search.SearchedRecipe

class SearchedRecipesAdapter : ListAdapter<SearchedRecipe, SearchedRecipesAdapter.RecipeViewHolder>(DIFF_CALLBACK) {

    override fun onCreateViewHolder(parent: ViewGroup,
                                    viewType: Int): RecipeViewHolder {
        val binding = SearchedRecipeItemBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return RecipeViewHolder(binding)
    }

    override fun onBindViewHolder(holder: RecipeViewHolder, position: Int) {
        val recipe = getItem(position)
        holder.bind(recipe)
    }

    inner class RecipeViewHolder(private val mRecipeBinding: SearchedRecipeItemBinding) : RecyclerView.ViewHolder(mRecipeBinding.root) {
        fun bind(recipe: SearchedRecipe?) {
            mRecipeBinding.recipe = recipe
        }

    }

    companion object {
        private val DIFF_CALLBACK: DiffUtil.ItemCallback<SearchedRecipe> = object : DiffUtil.ItemCallback<SearchedRecipe>() {
            override fun areItemsTheSame(oldItem: SearchedRecipe, newItem: SearchedRecipe): Boolean {
                return oldItem.id == newItem.id
            }

            @SuppressLint("DiffUtilEquals")
            override fun areContentsTheSame(oldItem: SearchedRecipe, newItem: SearchedRecipe): Boolean {
                return oldItem === newItem
            }
        }
    }
}